### PR TITLE
Device api doc typo fixes

### PR DESCRIPTION
**Pull request by lillian542**
fixes a few typos

### CODE REVIEW

File: pennylane/devices/experimental/device_api.py [hunk 1]
```diff
@@ -61,7 +61,7 @@ class Device(abc.ABC):
     .. details::
         :title: Porting from the old interface
 
-        :meth:`~.Device.batch_execute` and :meth:`~.Device.execute` are now a single method, :meth:`~.Device.execute`
+        :meth:`~.Device.batch_execute` and :meth:`~pennylane.Device.execute` are now a single method, :meth:`~.Device.execute`
```
Comment (albi3ro) [inline pennylane/devices/experimental/device_api.py:64]:
```suggestion
        :meth:`pennylane.Device.batch_execute` and :meth:`~pennylane.Device.execute` are now a single method, :meth:`~.Device.execute`
```
Probably good to do this for both references to the old device.

File: pennylane/devices/experimental/device_api.py [hunk 2]
```diff
@@ -362,7 +362,7 @@ def compute_derivatives(
         Returns:
             Tuple: The jacobian for each trainable parameter
 
-        See also :meth:`~.supports_differentiation` and :meth:`~.execute_and_compute_derivatives`.
+        See also :meth:`~.supports_derivatives` and :meth:`~.execute_and_compute_derivatives`.
```
Comment (timmysilv) [inline pennylane/devices/experimental/device_api.py:365]:
```suggestion
        .. seealso:: :meth:`~.supports_derivatives` and :meth:`~.execute_and_compute_derivatives`.
```

**Review by timmysilv — COMMENTED** [review]
I can't recommend line changes everywhere but I would also use the `seealso` directive in `execute_and_compute_jvp` and `execute_and_compute_vjp`

**Review by albi3ro — APPROVED** [review]
Thanks :)

**Review by timmysilv — APPROVED** [review]

**Pull request merged**